amf-graphql/shared/src/main/scala/amf/graphql/internal/spec/parser/syntax/GraphQLASTParserHelper.scala

```diff
@@ -198,4 +200,23 @@ trait GraphQLASTParserHelper extends AntlrASTParserHelper {
     shape.unresolved(typeName, Nil, Some(element.location))
     shape
   }
+
+  private def checkNameIsValid(name: String, location: Annotations)(implicit
+      ctx: GraphQLWebApiContext
+  ): Unit = {
+    if (isIntrospectionName(name)) invalidIntrospectionName(name, location)
+  }
+
+  private def isIntrospectionName(name: String) = name.startsWith("__")
+
+  private def invalidIntrospectionName(name: String, location: Annotations)(implicit
+      ctx: GraphQLWebApiContext
+  ): Unit = {
+    ctx.eh.violation(
+      InvalidIntrospectionName,
+      name,
```

Comment:
The 'target' field in an AMFValidationResult is the element's id, not the value of the evaluated property. It is something like this:
```
  Target: file://amf-cli/shared/src/test/resources/production/inter-spec-refs/oas-raml-securityScheme/api.json#/declares/scheme/oauth20
  Property: http://a.ml/vocabularies/security#type
```

Why can't this go as a custom shacl function instead of a parsing side validation ?